x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rules_list_snooze_panel.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import moment from 'moment';
+import { RuleTableItem } from '../../../../types';
+import {
+  SnoozePanel,
+  futureTimeToInterval,
+  usePreviousSnoozeInterval,
+  SnoozeUnit,
+  isRuleSnoozed,
+} from './rule_status_dropdown';
+
+const EMPTY_HANDLER = () => {};
+
+export interface RulesListSnoozePanelProps {
+  rule: RuleTableItem;
+  previousSnoozeInterval?: string | null;
+  onLoading?: (isLoading: boolean) => void;
+  onRuleChanged: () => Promise<void>;
+  onClose: () => void;
+  snoozeRule: (snoozeEndTime: string | -1, interval: string | null) => Promise<void>;
+  unsnoozeRule: () => Promise<void>;
+}
+
+export const RulesListSnoozePanel = (props: RulesListSnoozePanelProps) => {
+  const {
+    rule,
+    previousSnoozeInterval: propsPreviousSnoozeInterval,
+    onRuleChanged,
+    onClose,
+    snoozeRule,
+    unsnoozeRule,
+    onLoading = EMPTY_HANDLER,
+  } = props;
+
+  const { isSnoozedUntil } = rule;
+
+  const [previousSnoozeInterval, setPreviousSnoozeInterval] = usePreviousSnoozeInterval(
+    propsPreviousSnoozeInterval
+  );
+
+  const isSnoozed = useMemo(() => {
+    return isRuleSnoozed(rule);
+  }, [rule]);
+
+  const snoozeRuleAndStoreInterval = useCallback(
+    (newSnoozeEndTime: string | -1, interval: string | null) => {
+      if (interval) {
+        setPreviousSnoozeInterval(interval);
+      }
+      return snoozeRule(newSnoozeEndTime, interval);
+    },
+    [setPreviousSnoozeInterval, snoozeRule]
+  );
+
+  const onChangeSnooze = useCallback(
+    async (value: number, unit?: SnoozeUnit) => {
+      onLoading(true);
+      onClose();
+      try {
+        if (value === -1) {
+          await snoozeRuleAndStoreInterval(-1, null);
```

Comment:
I think we should wrap our async call in a try/catch and show a toaster with the error or a success one. Same remarks for all the async call.

Reply:
sounds good